### PR TITLE
Capture app, job and databricks parameters in spark streaming spans

### PR DESCRIPTION
**Pull request by paul-laffon-dd**
# What Does This Do

Capture spark application, job and databricks parameters in spark streaming spans

Those parameters are currently missing on streaming spans, preventing to link those back to databricks tasks

### CODE REVIEW

**Review by raphaelgavache — APPROVED** [review]
nice

File: dd-java-agent/instrumentation/spark/src/main/java/datadog/trace/instrumentation/spark/DatadogSparkListener.java [hunk 1]
```diff
@@ -168,6 +164,50 @@ public synchronized void finishApplication(
...
+    // In databricks, there is no application span. Adding the spark conf parameters to the top
+    // level spark span
+    captureApplicationParameters(builder);
+
+    if (properties != null) {
+      String databricksJobId = (String) properties.get("spark.databricks.job.id");
+      String databricksJobRunId = getDatabricksJobRunId(properties, databricksClusterName);
+
+      // spark.databricks.job.runId is the runId of the task, not of the Job
+      String databricksTaskRunId = (String) properties.get("spark.databricks.job.runId");
```
Comment (PerfectSlayer) [inline dd-java-agent/instrumentation/spark/src/main/java/datadog/trace/instrumentation/spark/DatadogSparkListener.java:193]:
```suggestion
      String databricksTaskRunId = properties.getProperty("spark.databricks.job.runId");
```

File: dd-java-agent/instrumentation/spark/src/main/java/datadog/trace/instrumentation/spark/DatadogSparkListener.java [hunk 2]
```diff
@@ -168,6 +164,50 @@ public synchronized void finishApplication(
...
+  }
+
+  private void addDatabricksSpecificTags(
+      AgentTracer.SpanBuilder builder, Properties properties, boolean withParentContext) {
+    // In databricks, there is no application span. Adding the spark conf parameters to the top
+    // level spark span
+    captureApplicationParameters(builder);
+
+    if (properties != null) {
+      String databricksJobId = (String) properties.get("spark.databricks.job.id");
```
Comment (PerfectSlayer) [inline dd-java-agent/instrumentation/spark/src/main/java/datadog/trace/instrumentation/spark/DatadogSparkListener.java:189]:
```suggestion
      String databricksJobId = properties.getProperty("spark.databricks.job.id");
```

**Review by PerfectSlayer — APPROVED** [review]
LGTM. Just minors changes related to the `Property` API.

**Pull request merged**